float32 x
float32 y
float32 z
float32 speed
int32 flag
float32 yaw
